### DISCUSSION TITLE
DataGrid problem

### DISCUSSION THREAD

**Discussion by isharrrry** [AvaloniaUI/Avalonia#9799] [Q&A]
DataGrid Items cannot be bound to DataTable or DataView.

The object property of DataGrid binding with {get; set;} will be written to null. It can only be solved by removing {get; set;}, whether IsOnlyRead of DataGrid is enabled or not.

DataGrid can only specify whether to display certain columns in AXAML, and cannot set and hide certain columns through annotations like [DisplayName ("***")]

**Comment by maxkatz6** [comment]
> DataGrid Items cannot be bound to DataTable or DataView.

It doesn't support these source types. You need to prepare items source and columns for the DataGrid instead.

> DataGrid can only specify whether to display certain columns in AXAML, and cannot set and hide certain columns through annotations like [DisplayName ("***")]

You mean, automatic columns generation? This can be improved, yes.

> The object property of DataGrid binding with {get; set;} will be written to null. It can only be solved by removing {get; set;}, whether IsOnlyRead of DataGrid is enabled or not.

Not sure if I understood.
